pkg/generators/backend/config/worker_options.go

```diff
@@ -1,57 +1,30 @@
-	opts := WorkerOptions{
-		RackEnv:                              &pod.ClearTextValue{Value: *spec.Config.RackEnv},
-		ConfigRedisProxy:                     &pod.ClearTextValue{Value: spec.Config.RedisStorageDSN},
-		ConfigRedisSentinelHosts:             &pod.ClearTextValue{Value: ""},
-		ConfigRedisSentinelRole:              &pod.ClearTextValue{Value: ""},
-		ConfigQueuesMasterName:               &pod.ClearTextValue{Value: spec.Config.RedisQueuesDSN},
-		ConfigQueuesSentinelHosts:            &pod.ClearTextValue{Value: ""},
-		ConfigQueuesSentinelRole:             &pod.ClearTextValue{Value: ""},
-		ConfigMasterServiceID:                &pod.ClearTextValue{Value: fmt.Sprintf("%d", *spec.Config.MasterServiceID)},
-		ConfigRedisAsync:                     &pod.ClearTextValue{Value: strconv.FormatBool(*spec.Worker.Config.RedisAsync)},
-		ConfigWorkersLoggerFormatter:         &pod.ClearTextValue{Value: *spec.Worker.Config.LogFormat},
-		ConfigWorkerPrometheusMetricsEnabled: &pod.ClearTextValue{Value: "true"},
-		ConfigWorkerPrometheusMetricsPort:    &pod.ClearTextValue{Value: "9421"},
-
-		ConfigEventsHook:             &pod.SecretValue{Value: spec.Config.SystemEventsHookURL},
-		ConfigEventsHookSharedSecret: &pod.SecretValue{Value: spec.Config.SystemEventsHookPassword},
-	}
+// NewWorkerOptions returns worker options for the given saasv1alpha1.BackedSpec
+func NewWorkerOptions(spec saasv1alpha1.BackendSpec) pod.Options {
+	opts := pod.Options{}
 
-	if spec.Config.ErrorMonitoringService != nil && spec.Config.ErrorMonitoringKey != nil {
-		opts.ConfigHoptoadService = &pod.SecretValue{Value: *spec.Config.ErrorMonitoringService}
-		opts.ConfigHoptoadAPIKey = &pod.SecretValue{Value: *spec.Config.ErrorMonitoringKey}
-	}
+	opts.Unpack(spec.Config.RackEnv).IntoEnvvar("RACK_ENV")
+	opts.Unpack(spec.Config.RedisStorageDSN).IntoEnvvar("CONFIG_REDIS_PROXY")
+	opts.Unpack("").IntoEnvvar("CONFIG_REDIS_SENTINEL_HOSTS")
+	opts.Unpack("").IntoEnvvar("CONFIG_REDIS_SENTINEL_ROLE")
+	opts.Unpack(spec.Config.RedisQueuesDSN).IntoEnvvar("CONFIG_QUEUES_MASTER_NAME")
+	opts.Unpack("").IntoEnvvar("CONFIG_QUEUES_SENTINEL_HOSTS")
+	opts.Unpack("").IntoEnvvar("CONFIG_QUEUES_SENTINEL_ROLE")
+	opts.Unpack(spec.Config.MasterServiceID).IntoEnvvar("CONFIG_MASTER_SERVICE_ID")
+	opts.Unpack(spec.Worker.Config.RedisAsync).IntoEnvvar("CONFIG_REDIS_ASYNC")
+	opts.Unpack(spec.Worker.Config.LogFormat).IntoEnvvar("CONFIG_WORKERS_LOGGER_FORMATTER")
+	opts.Unpack("true").IntoEnvvar("CONFIG_WORKER_PROMETHEUS_METRICS_ENABLED")
+	opts.Unpack("9421").IntoEnvvar("CONFIG_WORKER_PROMETHEUS_METRICS_PORT")
+	opts.Unpack(spec.Config.SystemEventsHookURL).IntoEnvvar("CONFIG_EVENTS_HOOK").AsSecretRef("backend-system-events-hook")
+	opts.Unpack(spec.Config.SystemEventsHookPassword).IntoEnvvar("CONFIG_EVENTS_HOOK_SHARED_SECRET").AsSecretRef("backend-system-events-hook")
+	opts.Unpack(spec.Config.ErrorMonitoringService).IntoEnvvar("CONFIG_HOPTOAD_SERVICE").AsSecretRef("backend-error-monitoring")
```

Comment:
this 2 monitoring key envvars addition with secret ref were optional, depending if the both the 2 values were empty in the CR

1. with the new format, it autodetects if any of the 2 values within the CR are set, the envvars will be added with their secret ref?

2 I've seen on zync something checking if empty `EmptyIf(!spec.Config.Bugsnag.Enabled())`, what's the difference?

Reply:
The default behavior of any option is to be unset if it's underlying value is nil (the field is not set in the custom resource). So in this case, if the error monitoring keys are not set, nor will be the envvars related to them.
The `EmptyIf()` function causes the envvar to be set with an empty value instead of not being present.

Reply:
thanks for the explanation!